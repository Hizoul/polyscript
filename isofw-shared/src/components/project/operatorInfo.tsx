import { useEdit } from "isofw-shared/src/util/xpfwdata"
import { FormStore } from "isofw-shared/src/util/xpfwform"
import {
  OperatorRelation, ProjectCameras, ProjectForm,
  ProjectOperatorCameraMapping, ProjectOperators, ProjectProgram, ShotCamera, ShotPreset
} from "isofw-shared/src/xpfwDefs/project"
import { ProjectShot } from "isofw-shared/src/xpfwDefs/project"
import { find, get } from "lodash"

const currentOperatorKey = `current${ProjectOperators.title}`
const whichViewIsActiveKey = `operatorInView`
const isOperatorChooserVisibleKey = `operatorChooserVisible`

const changeValue = (keyToUse: string, prefix?: string, toggleChooser?: boolean) => {
  return (operator: any) => {
    return () => {
      FormStore.setValue(keyToUse, operator, prefix)
      if (toggleChooser) {
        FormStore.setValue(isOperatorChooserVisibleKey, false, prefix)
      }
    }
  }
}

const useOperatorInfo = (id: string, mapTo?: string, prefix?: string, reset?: boolean, defItem?: any) => {
  const editHelper = useEdit(id, ProjectForm, mapTo, prefix, reset)
  const currentOperator = FormStore.getValue(currentOperatorKey, prefix)
  const item = get(editHelper, "original", defItem)
  const mappings: any = find(get(item, String(ProjectOperatorCameraMapping.title), []),
    [OperatorRelation.title, currentOperator])
  const currentCameras = mappings && mappings[String(ProjectCameras.title)] ?
    mappings[String(ProjectCameras.title)] : []
  const program = get(item, String(ProjectProgram.title), [])
  let filteredList = program
  const presetByCameraObj: any = {}
  filteredList.forEach((subItem: any) => {
    if (presetByCameraObj[subItem[String(ShotCamera.title)]] == null) {
      presetByCameraObj[subItem[String(ShotCamera.title)]] = []
    }
    presetByCameraObj[subItem[String(ShotCamera.title)]].push(subItem[String(ShotPreset.title)])
  })
  const presetByCamera: any[] = []
  for (const k of Object.keys(presetByCameraObj)) {
    presetByCamera.push({
      camera: k, presets: presetByCameraObj[k]
    })
  }
  if (currentCameras.length > 0) {
    filteredList = filteredList.filter(
      (subItem: any) => currentCameras.indexOf(subItem[String(ShotCamera.title)]) !== -1)
  }
  const currentPreset = program[get(item, String(ProjectShot.title), 0)]
  return {
    ...editHelper,
    currentOperator,
    currentPreset,
    currentCameras,
    presetByCamera,
    filteredList,
    isPresetView: FormStore.getValue(whichViewIsActiveKey, prefix) === 1,
    isOperatorChooserVisible: FormStore.getValue(isOperatorChooserVisibleKey, prefix) === true,
    changeOperator: changeValue(currentOperatorKey, prefix, true),
    useScriptView: changeValue(whichViewIsActiveKey, prefix)(0),
    usePresetView: changeValue(whichViewIsActiveKey, prefix)(1),
    showOperatorChooser: changeValue(isOperatorChooserVisibleKey, prefix)(true),
    hideOperatorChooser: changeValue(isOperatorChooserVisibleKey, prefix)(false)
  }
}

export default useOperatorInfo
export {
  changeValue, currentOperatorKey, whichViewIsActiveKey
}
