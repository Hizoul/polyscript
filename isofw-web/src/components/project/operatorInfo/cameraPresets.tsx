import { DbStore, IFormEditProps, IFormShowProps, SharedFormEdit } from "@xpfw/ui-shared"
import { BlockTitle, Card, CardFooter, CardHeader, Link } from "framework7-react"
import sharedPresetUpdater, { PresetUpdaterProps } from "isofw-shared/src/components/preset/updater"
import val from "isofw-shared/src/globals/val"
import { PresetNumberField } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectForm, ProjectName, ProjectProgram, ProjectShot, ShotCamera,
  ShotDuration, ShotMovement, ShotMovementTowards, ShotName,
  ShotPreset, ShotRemarksDirector, ShotRemarksOperator, ShotType } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import NameDisplayer from "isofw-web/src/components/displayName"
import { get } from "lodash"
import * as React from "react"

const PresetCard: React.FunctionComponent<PresetUpdaterProps> = (props) => {
  return (
    <Card>
      <CardHeader>
        <span>Preset #{get(props.item, PresetNumberField.mapTo)}</span>
      </CardHeader>
      <CardFooter>
        <Link text="Update" />
        <Link
          text={props.isReady ? "un-ready" : "ready"}
          onClick={props.isReady ? props.setNotReady : props.setReady}
        />
      </CardFooter>
    </Card>
  )
}
const WrappedPresetCard = sharedPresetUpdater(PresetCard)
const CameraPresets: React.FunctionComponent<IFormShowProps> = (props) => {
  return (
    <div>
      <BlockTitle>
        <NameDisplayer collection={val.service.camera} id={get(props.item, "camera")} getNameFrom={ProjectName.mapTo} placeholder="" />
      </BlockTitle>
      <div className="presetCards">
        {get(props.item, "presets", []).map((preset: any) => <WrappedPresetCard key={preset} id={preset} />)}
      </div>
    </div>
  )
}

export default CameraPresets
export {
  PresetCard
}
