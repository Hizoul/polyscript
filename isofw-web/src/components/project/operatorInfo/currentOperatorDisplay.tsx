import { MailField } from "@xpfw/data"
import { Link, List, ListItem, Popover } from "framework7-react"
import val from "isofw-shared/src/globals/val"
import { OperatorRelation, ProjectCameras, ProjectName,
  ProjectOperatorCameraMapping } from "isofw-shared/src/xpfwDefs/project"
import NameDisplayer from "isofw-web/src/components/displayName"
import I18n from "isofw-web/src/components/i18n"
import { get } from "lodash"
import * as React from "react"

const CurrentOperatorDisplay: React.FunctionComponent<any> = (props) => {
  const name = props.currentOperator && props.currentOperator.length > 0 ?
  <NameDisplayer collection={val.service.user} id={props.currentOperator} getNameFrom={String(MailField.title)} /> : "No one"
  return (
    <div>
      <Link popoverOpen=".operatorChooser" onClick={props.showOperatorChooser} className="subtitleBox centerText">
        <div className="subtitle">{name}</div>
        {props.currentCameras && props.currentCameras.length > 0 ? (
          <span>CA {props.currentCameras.map((camera: any) =>
            <NameDisplayer key={camera} collection={val.service.camera} id={camera} getNameFrom={String(ProjectName.title)} />)}</span>
        ) : <I18n text="operator.cameras.showAll" />}
      </Link>
      <Popover className="operatorChooser" opened={props.isOperatorChooserVisible}>
        <List>
          <ListItem onClick={props.changeOperator("")} popoverClose=".operatorChooser">
            <div slot="title"><I18n text="operator.cameras.all" />
            </div>
            <div slot="footer">
              <I18n text="operator.cameras.allCameras" />
            </div>
          </ListItem>
          {get(props.item, String(ProjectOperatorCameraMapping.title), []).map((mapping: any) => {
            const cameras = mapping[String(ProjectCameras.title)]
            return (
              <ListItem key={mapping[String(OperatorRelation.title)]} onClick={props.changeOperator(mapping[String(OperatorRelation.title)])} popoverClose={true}>
                <div slot="title">
                  <NameDisplayer collection={val.service.user} id={mapping[String(OperatorRelation.title)]} getNameFrom={String(MailField.title)} />
                </div>
                <div slot="footer">
                  {cameras && cameras.length > 0 ? (
                    <span><I18n text="operator.cameras.list" />{cameras.map((camera: any) =>
                      <NameDisplayer key={camera} collection={val.service.camera} id={camera} getNameFrom={String(ProjectName.title)} />)}
                    </span>
                  ) : undefined}
                </div>
              </ListItem>
          )})}
        </List>
      </Popover>
    </div>
  )
}

export default CurrentOperatorDisplay
