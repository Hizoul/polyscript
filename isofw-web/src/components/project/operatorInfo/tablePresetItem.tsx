import { DbStore, IFormEditProps, IFormShowProps, SharedFormEdit } from "@xpfw/ui-shared"
import val from "isofw-shared/src/globals/val"
import { PresetNumberField } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectForm, ProjectName, ProjectProgram, ProjectShot, ShotCamera,
  ShotDuration, ShotImportance, ShotMovement, ShotMovementTowards,
  ShotName, ShotPreset, ShotRemarksDirector, ShotRemarksOperator, ShotType } from "isofw-shared/src/xpfwDefs/project"
import NameDisplayer from "isofw-web/src/components/displayName"
import { get } from "lodash"
import * as React from "react"

const TablePresetItem: React.FunctionComponent<IFormShowProps & {index: number}> = (props) => {
  let importance = ""
  const importanceValue = get(props.item, ShotImportance.mapTo)
  if (importanceValue === "r") {
    importance = "!"
  } else if (importanceValue === "m") {
    importance = "!!"
  }
  return (
    <tr id={`presetPositioner${props.index}`}>
      <td>{props.index}</td>
      <td>
        <NameDisplayer collection={val.service.camera} id={get(props.item, ShotCamera.mapTo)} getNameFrom={ProjectName.mapTo} placeholder="" />
      </td>
      <td>
        <NameDisplayer collection={val.service.preset} id={get(props.item, ShotPreset.mapTo)} getNameFrom={PresetNumberField.mapTo} placeholder="" />
      </td>
      <td>{importance}</td>
      <td>{get(props.item, ShotName.mapTo)}</td>
      <td>{get(props.item, ShotType.mapTo)}</td>
      <td>{get(props.item, ShotMovement.mapTo)}</td>
      <td>{get(props.item, ShotMovementTowards.mapTo)}</td>
      <td>{get(props.item, ShotDuration.mapTo)}</td>
      <td>placeholder</td>
      <td>
        {get(props.item, ShotRemarksDirector.mapTo)}<br />
        {get(props.item, ShotRemarksOperator.mapTo)}
      </td>
    </tr>
  )
}

export default TablePresetItem
