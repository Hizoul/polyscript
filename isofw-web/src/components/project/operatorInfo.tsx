import * as React from "react"
import { SharedFormEdit, IFormEditProps, DbStore, IFormShowProps } from "@xpfw/ui-shared";
import { ProjectForm, ProjectShot, ProjectProgram, ProjectName, ShotName, ShotType, ShotMovement, ShotMovementTowards, ShotDuration, ShotRemarksDirector, ShotRemarksOperator } from "isofw-shared/src/xpfwDefs/project";
import { FormStore, SharedField } from "@xpfw/form-shared";
import LoadingPage from "../loading";
import { get } from "lodash"
import { ComponentRegistry } from "@xpfw/form-shared"
import { FieldType } from "isofw-shared/src/util/xpfwvalidate";
import WebButton from "isofw-web/src/components/button";
import { Row, Col, Block, Link, Icon } from "framework7-react";
import "./style.sass"
import urls from "isofw-shared/src/globals/url";

const PresetItem: React.FunctionComponent<IFormShowProps> = (props) => {
  return (
    <tr>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ShotType.mapTo)}</td>
      <td>{get(props.item, ShotName.mapTo)}</td>
      <td>{get(props.item, ShotMovement.mapTo)}</td>
      <td>{get(props.item, ShotMovementTowards.mapTo)}</td>
      <td>{get(props.item, ShotDuration.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>
        {get(props.item, ShotRemarksDirector.mapTo)}<br />
        {get(props.item, ShotRemarksOperator.mapTo)}
      </td>
    </tr>
  )
}

const OperatorInfo: React.FunctionComponent<IFormEditProps> = (props) => {
  console.log(" operator props ", props, get(props.original, ProjectProgram.mapTo, []))
  let i = 0
  const items = get(props.original, `result.${ProjectProgram.mapTo}`, []).map((item: any) => <PresetItem loading={false} item={item} key={++i} />)
  return (
    <div>
      <Row>
        <Col>
            <div className="subtitleBox withPadding">
              <Link href={urls.projectOverview} className="flex">
                <div className="flex1 column">
                  <div className="subtitle">active project</div>
                  {get(props, "original.result.name")}
                </div>
                <Icon fa="th-list" />
              </Link>
            </div>
        </Col>
        <Col>
          <div className="subtitleBox centerText">
            <div className="subtitle">Operator info</div>
            CA 5 6 9
          </div>
        </Col>
        <Col>
          <div className="subtitleBox withPadding">
            <div className="subtitle">Current shot</div>
            5
          </div>
        </Col>
      </Row>
      <Row className="marginTopBottom">
        <Col>
          <WebButton big fill text="shot ready" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="update preset" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="wide" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="stop zoom" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="tele" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="not ready" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="following position" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="script view" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="preset view" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
      </Row>
      <div className="data-table card" style={{marginLeft: "0pt", marginRight: "0pt"}}>
        <table>
          <thead>
            <tr>
              <th>shot</th>
              <th>CA</th>
              <th>Preset</th>
              <th>Type</th>
              <th>Object</th>
              <th>Movement</th>
              <th>Towards</th>
              <th>Duration</th>
              <th>picture</th>
              <th>remarks</th>
            </tr>
          </thead>
          <tbody>
            {items}
          </tbody>
        </table>
      </div>
      {props.loading ? <LoadingPage /> : null}
    </div>
  )
}

export default SharedFormEdit<{}>(OperatorInfo)