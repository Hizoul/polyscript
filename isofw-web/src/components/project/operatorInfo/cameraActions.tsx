import { Col,  Row } from "framework7-react"
import useCameraActions from "isofw-shared/src/components/project/cameraActions"
import { ShotCamera } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import { observer } from "mobx-react-lite"
import * as React from "react"

const CameraActions: React.FunctionComponent<any> = observer((props) => {
  const actionHelper = useCameraActions(ShotCamera, undefined, props.prefix)
  return (
    <Row className="marginTopBottom">
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.ready" iconFa="check" onClick={actionHelper.setReady} loading={actionHelper.loading} />
      </Col>
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.update" iconFa="check" onClick={actionHelper.savePreset} loading={actionHelper.loading} />
      </Col>
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.wide" iconFa="check" onClick={actionHelper.wideZoom} loading={actionHelper.loading} />
      </Col>
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.stopZoom" iconFa="check" onClick={actionHelper.stopZoom} loading={actionHelper.loading} />
      </Col>
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.tele" iconFa="check" onClick={actionHelper.teleZoom} loading={actionHelper.loading} />
      </Col>
      <Col>
        <WebButton large={true} fill={true} text="operator.btns.notReady" iconFa="check" onClick={actionHelper.setNotReady} loading={actionHelper.loading} />
      </Col>
      {/* <Col>
        <WebButton large={true} fill={true} text="operator.btns.autoScroll" iconFa="check" onClick={actionHelper.submitEdit} loading={actionHelper.loading} />
      </Col> */}
      {props.children}
    </Row>
  )
})

export default CameraActions
