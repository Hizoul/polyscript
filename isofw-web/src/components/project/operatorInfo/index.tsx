import { Col, Icon, Link, Row } from "framework7-react"
import useOperatorInfo from "isofw-shared/src/components/project/operatorInfo"
import urls from "isofw-shared/src/globals/url"
import val from "isofw-shared/src/globals/val"
import { IEditHookProps } from "isofw-shared/src/util/xpfwdata"
import { ProjectName, ProjectShot, ShotCamera, ShotName } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import NameDisplayer from "isofw-web/src/components/displayName"
import { get } from "lodash"
import { observer } from "mobx-react-lite"
import * as React from "react"
import LoadingPage from "../../loading"
import "../style.sass"
import CameraPresets from "./cameraPresets"
import CurrentOperatorDisplay from "./currentOperatorDisplay"
import TablePresetItem from "./tablePresetItem"

const OperatorInfo: React.FunctionComponent<IEditHookProps> = observer((props) => {
  const previousPosition = React.useState(0)
  const operatorHelper = useOperatorInfo(props.id, props.mapTo, props.prefix)
  React.useEffect(() => {
    const newPosition = get(operatorHelper.original, String(ProjectShot.title))
    if (newPosition !== previousPosition[0]) {
      previousPosition[1](newPosition)
      const element = document.getElementById(`presetPositioner${newPosition}`)
      if (element != null) {
        element.scrollIntoView({
          behavior: "smooth", block: "center", inline: "center"
        })
      }
    }
  })
  let i = 0
  const content = operatorHelper.isPresetView ? (
    <div>
      {operatorHelper.presetByCamera.map((item: any) => {
            i++
            if (operatorHelper.currentCameras.length > 0 &&
              operatorHelper.currentCameras.indexOf(item.camera) === -1) {return null}
            return <CameraPresets item={item} key={item.camera} />
          })}
    </div>
  ) : (
    <div className="data-table card" style={{marginLeft: "0pt", marginRight: "0pt"}}>
      <table>
        <thead>
          <tr>
            <th>shot</th>
            <th>CA</th>
            <th>Preset</th>
            <th>!</th>
            <th>name</th>
            <th>Type</th>
            <th>Movement</th>
            <th>Towards</th>
            <th>Duration</th>
            <th>picture</th>
            <th>remarks</th>
          </tr>
        </thead>
        <tbody>
          {operatorHelper.filteredList.map((item: any) => {
            i++
            return <TablePresetItem item={item} key={i} />
          })}
        </tbody>
      </table>
    </div>
  )
  return (
    <div>
      <Row>
        <Col>
            <div className="subtitleBox withPadding">
              <Link href={urls.projectOverview} className="flex">
                <div className="flex1 column">
                  <div className="subtitle">active project</div>
                  {get(operatorHelper, "original.name")}
                </div>
                <Icon fa="th-list" />
              </Link>
            </div>
        </Col>
        <Col>
          <CurrentOperatorDisplay {...props} {...operatorHelper} />
        </Col>
        <Col>
          <div className="subtitleBox withPadding">
            <div className="subtitle">Current shot</div>
            {operatorHelper.currentPreset ? (
              <div>
                <NameDisplayer
                  collection={val.service.camera}
                  id={get(operatorHelper.currentPreset, String(ShotCamera.title))}
                  getNameFrom={String(ProjectName.title)}
                  placeholder=""
                /> &nbsp;
                {get(operatorHelper.currentPreset, String(ShotName.title), "")}
              </div>
            ) : null}
          </div>
        </Col>
      </Row>
      <Row className="marginTopBottom">
        <Col>
          <WebButton large={true} fill={true} text="shot ready" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="update preset" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="wide" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="stop zoom" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="tele" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="not ready" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="following position" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton
            large={true}
            fill={true}
            text={operatorHelper.isPresetView ? "script view" : "preset view"}
            iconFa={operatorHelper.isPresetView ? "list-ol" : "images"}
            onClick={operatorHelper.isPresetView ? operatorHelper.useScriptView : operatorHelper.usePresetView}
          />
        </Col>
      </Row>
      {content}
      {operatorHelper.loading ? <LoadingPage /> : null}
    </div>
  )
})
export default OperatorInfo
