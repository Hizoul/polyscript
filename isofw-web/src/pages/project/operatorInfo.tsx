import { ProjectForm, ProjectShot } from "isofw-shared/src/xpfwDefs/project"
import "isofw-web/src/components/form"
import WebPageContainer from "isofw-web/src/components/pageContainer"
import OperatorInfo from "isofw-web/src/components/project/operatorInfo"
import { get } from "lodash"
import * as React from "react"

const OperatorInfoPage: React.FunctionComponent<any> = (props) => {
  const id = get(props, "id")
  return (
    <WebPageContainer backLink={true} requireLoggedIn={true} name="operatorInfo" title="Operator info's">
      <OperatorInfo schema={{}} id={id} reset={true} prefix="operatorInfo" />
    </WebPageContainer>
  )
}

export default OperatorInfoPage
