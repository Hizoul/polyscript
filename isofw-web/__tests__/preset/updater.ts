import presetUpdaterTest from "isofw-shared/src/tests/preset/updater"
import { PresetCard } from "isofw-web/src/components/project/operatorInfo/cameraPresets"

presetUpdaterTest(PresetCard)
