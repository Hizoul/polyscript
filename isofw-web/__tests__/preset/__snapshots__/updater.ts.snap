// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preset Update works as planned:  after setting 2 falls  1`] = `
<div
  className="Card"
>
  {}
  <div
    className="CardHeader"
  >
    {}
    <span>
      Preset #
    </span>
  </div>
  <div
    className="CardFooter"
  >
    {}
    <div
      className="Link"
    >
      {
  "text": "Update"
}
    </div>
    <div
      className="Link"
    >
      {
  "text": "ready"
}
    </div>
  </div>
</div>
`;

exports[`Preset Update works as planned:  after setting to true  1`] = `
<div
  className="Card"
>
  {}
  <div
    className="CardHeader"
  >
    {}
    <span>
      Preset #
    </span>
  </div>
  <div
    className="CardFooter"
  >
    {}
    <div
      className="Link"
    >
      {
  "text": "Update"
}
    </div>
    <div
      className="Link"
    >
      {
  "text": "ready"
}
    </div>
  </div>
</div>
`;

exports[`Preset Update works as planned: Before anything 1`] = `
<div
  className="Card"
>
  {}
  <div
    className="CardHeader"
  >
    {}
    <span>
      Preset #
    </span>
  </div>
  <div
    className="CardFooter"
  >
    {}
    <div
      className="Link"
    >
      {
  "text": "Update"
}
    </div>
    <div
      className="Link"
    >
      {
  "text": "ready"
}
    </div>
  </div>
</div>
`;
